### PR TITLE
feat: Error handling

### PR DESCRIPTION
**Pull request by rjsturing**
# Description
Implements error handling into the fetch call

# Screenshot(s)
Screenshot 2023-12-02 at 6 48 46 PM

# Checklist
- [x] My PR has an appropriately descriptive and concise title.
- [x] My code follows the Turing Style Guides and best practices.
- [x] I ran the code locally and verified that there are no visible errors.
- [x] feat: My PR clearly describes what feature I'm adding and any changes needed to make it work.

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -6,13 +6,22 @@ import MovieDetail from "../MovieDetail/MovieDetail";
...
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(`Server responded with status ${response.status}`);
+        }
+        return response.json();
+      })
+      .then((data) => setMovies(data.movies))
+      .catch((error) => {
+        console.error(error.message);
+        setError(`Server error: ${error.message}`);
```
Comment (ericahagle) [inline src/App/App.js:22]: @rjsturing Maybe we can make this messaging a little friendlier for users. Something like, "Oopsie! Something went wrong. Please try again later." ?

**Review by ericahagle — CHANGES_REQUESTED** [review]

File: src/App/App.js (hunk 1 above)
Comment (rjsturing) [reply src/App/App.js]: Great idea, the user should have a more clear idea of what isn't working correctly. Completed!

Comment (rjsturing) [reply src/App/App.js]: Great idea, the user should have a more clear idea of what isn't working correctly. Completed!

**Review by ericahagle — APPROVED** [review]
Beautiful! Great work once again, @rjsturing! 🌮

**Pull request merged**